gperftools: can't be installed in mac OS (blocking mongodb)

**Describe the bug**
A clear and concise description of what the bug is.
I can't install mongodb in mac OS. I get the following errors when using nix shell

```
checking how clang reports undeclared, standard C functions... configure: error: in `/private/tmp/nix-build-gperftools-2.8.drv-0/gperftools-2.8':
configure: error: cannot detect from compiler exit status or warnings
See `config.log' for more details
builder for '/nix/store/31v0wys0dz9clxyvjx3lc3jrnwnl3iqh-gperftools-2.8.drv' failed with exit code 1
cannot build derivation '/nix/store/zgxzf15h0dbbp9rr04kklgaflqk6pd20-mongodb-3.4.24.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/z25i9ykzh4s75l6q1dffqyvdl9nnyckk-lorri-keep-env-hack-nix-shell.drv': 1 dependencies couldn't be built
error: build of '/nix/store/z25i9ykzh4s75l6q1dffqyvdl9nnyckk-lorri-keep-env-hack-nix-shell.drv' failed
```

and the following if I do a vanilla install `nix-env -i mongodb`
```
checking how clang reports undeclared, standard C functions... configure: error: in `/private/tmp/nix-build-gperftools-2.8.drv-0/gperftools-2.8':
configure: error: cannot detect from compiler exit status or warnings
See `config.log' for more details
builder for '/nix/store/31v0wys0dz9clxyvjx3lc3jrnwnl3iqh-gperftools-2.8.drv' failed with exit code 1
cannot build derivation '/nix/store/zgxzf15h0dbbp9rr04kklgaflqk6pd20-mongodb-3.4.24.drv': 1 dependencies couldn't be built

```


**To Reproduce**
try installing mongodb by `nix-env -i mongodb` and observe the error.
**Expected behavior**
Should be able to install mongodb on mac os. 

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Additional context**
Add any other context about the problem here.

**Notify maintainers**
@kevincox @mohe2015 @timokau @dasJ @Mic92 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.2.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixpkgs-20.09pre236005.2b7c0dcdaab"`
 - channels(ashkanaleali): `"darwin, home-manager, nixpkgs-21.03pre264143.7d71001b796, nixpkgs-unstable-21.03pre264143.7d71001b796"`
 - nixpkgs: `/Users/ashkanaleali/.nix-defexpr/channels/nixpkgs`

--- a/pkgs/development/libraries/gperftools/default.nix
+++ b/pkgs/development/libraries/gperftools/default.nix
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, fetchpatch, autoreconfHook, libunwind }:
+{ stdenv, lib, fetchurl, fetchpatch, autoreconfHook, libunwind }:
 
 stdenv.mkDerivation rec {
   name = "gperftools-2.8";
@@ -20,20 +20,20 @@ stdenv.mkDerivation rec {
   nativeBuildInputs = [ autoreconfHook ];
 
   # tcmalloc uses libunwind in a way that works correctly only on non-ARM linux
-  buildInputs = stdenv.lib.optional (stdenv.isLinux && !(stdenv.isAarch64 || stdenv.isAarch32)) libunwind;
+  buildInputs = lib.optional (stdenv.isLinux && !(stdenv.isAarch64 || stdenv.isAarch32)) libunwind;
 
   # Disable general dynamic TLS on AArch to support dlopen()'ing the library:
   # https://bugzilla.redhat.com/show_bug.cgi?id=1483558
-  configureFlags = stdenv.lib.optional (stdenv.isAarch32 || stdenv.isAarch64)
+  configureFlags = lib.optional (stdenv.isAarch32 || stdenv.isAarch64)
     "--disable-general-dynamic-tls";
 
-  prePatch = stdenv.lib.optionalString stdenv.isDarwin ''
+  prePatch = lib.optionalString stdenv.isDarwin ''
     substituteInPlace Makefile.am --replace stdc++ c++
     substituteInPlace Makefile.in --replace stdc++ c++
     substituteInPlace libtool --replace stdc++ c++
   '';
 
-  NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin
+  NIX_CFLAGS_COMPILE = lib.optionalString stdenv.isDarwin
     "-D_XOPEN_SOURCE";
 
   # some packages want to link to the static tcmalloc_minimal
@@ -42,7 +42,7 @@ stdenv.mkDerivation rec {
 
   enableParallelBuilding = true;
 
-  meta = with stdenv.lib; {
+  meta = with lib; {
     homepage = "https://github.com/gperftools/gperftools";
     description = "Fast, multi-threaded malloc() and nifty performance analysis tools";
     platforms = with platforms; linux ++ darwin;

--- a/pkgs/development/libraries/gperftools/default.nix
+++ b/pkgs/development/libraries/gperftools/default.nix
@@ -34,7 +34,7 @@ stdenv.mkDerivation rec {
   '';
 
   NIX_CFLAGS_COMPILE = stdenv.lib.optionalString stdenv.isDarwin
-    "-D_XOPEN_SOURCE -Wno-aligned-allocation-unavailable";
+    "-D_XOPEN_SOURCE";
 
   # some packages want to link to the static tcmalloc_minimal
   # to drop the runtime dependency on gperftools

